File: JavaScript/tests/1.test.js
import { twoSum } from "../src/1.js";
import { assertEquals } from "jsr:@std/assert";

const cases = [
	{
		nums: [2, 7, 11, 15],
		target: 9,
		want: [0, 1],
	},
	{
		nums: [3, 2, 4],
		target: 6,
		want: [1, 2],
	},
	{
		nums: [3, 3],
		target: 6,
		want: [0, 1],
	},
];

Deno.test("1. Two Sum", () => {
	for (let i = 0; i < cases.length; i++) {
		const { nums, target, want } = cases[i];
		const result = twoSum(nums, target);
		assertEquals(result, want, `case ${i + 1}`);
	}
});

File: C++/tests/1_test.cpp
#include <gtest/gtest.h>
#include <solution_tmpl.h>
using namespace std;

class TwoSumTest : public ::testing::Test {
protected:
  Solution solution;
};

TEST_F(TwoSumTest, FindsIndicesOfTwoSum) {
  vector<int> nums = {2, 7, 11, 15};
  int target = 9;
  vector<int> want = {0, 1};

  vector<int> actual = solution.twoSum(nums, target);
  EXPECT_EQ(actual, want);
}

TEST_F(TwoSumTest, FindsIndicesInDifferentOrder) {
  vector<int> nums = {3, 2, 4};
  int target = 6;
  vector<int> want = {1, 2};

  vector<int> actual = solution.twoSum(nums, target);
  EXPECT_EQ(actual, want);
}

TEST_F(TwoSumTest, HandlesDuplicateValues) {
  vector<int> nums = {3, 3};
  int target = 6;
  vector<int> want = {0, 1};

  vector<int> actual = solution.twoSum(nums, target);
  EXPECT_EQ(actual, want);
}
